sdk/python/pulumi_kubernetes/helm/v3/helm.py

```diff
@@ -555,91 +507,24 @@ def _parse_chart(all_config: Tuple[str, Union[ChartOpts, LocalChartOpts], pulumi
         fetch_opts_dict = {}
         if config.fetch_opts is not None:
             fetch_opts_dict = {k: v for k, v in vars(config.fetch_opts).items() if v is not None}
-        fetch_opts_dict["destination"] = chart_dir
         if config.version is not None:
             fetch_opts_dict["version"] = config.version
         fetch_opts = FetchOpts(**fetch_opts_dict)
-
-        # Fetch the chart.
-        _fetch(chart_to_fetch, fetch_opts)
-        # Sort the directories into alphabetical order, and choose the first
-        fetched_chart_name = sorted(os.listdir(chart_dir), key=str.lower)[0]
-        chart = os.path.join(chart_dir, fetched_chart_name)
+        config.fetch_opts = fetch_opts
     else:
         chart = config.path
 
-    default_values = os.path.join(chart, 'values.yaml')
+    config.release_name = release_name
 
-    # Write overrides file.
-    vals = config.values if config.values is not None else {}
-    data = pulumi.Output.from_input(vals).apply(lambda x: json.dumps(x))
-    file = open(overrides, 'w')
-    pulumi.Output.all(file, data).apply(_write_override_file)
-
-    namespace_arg = ['--namespace', config.namespace] if config.namespace else []
-    crd_arg = ['--include-crds'] if _is_helm_v3() else []
-
-    # Use 'helm template' to create a combined YAML manifest.
-    cmd = ['helm', 'template', chart, '--name-template', release_name,
-           '--values', default_values, '--values', overrides_filename]
-    cmd.extend(namespace_arg)
-    cmd.extend(crd_arg)
-
-    chart_resources = pulumi.Output.all(cmd, data).apply(_run_helm_cmd)
+    json_opts = json.dumps(config.toJson())
 
     # Rather than using the default provider for the following invoke call, use the version specified
     # in package.json.
     invoke_opts = pulumi.InvokeOptions(version=_utilities.get_version())
 
-    objects = chart_resources.apply(
-        lambda text: pulumi.runtime.invoke('kubernetes:yaml:decode', {
-            'text': text, 'defaultNamespace': config.namespace}, invoke_opts).value['result'])
+    objects = pulumi.runtime.invoke('kubernetes:helm:template', {
+        'jsonOpts': json_opts, 'defaultNamespace': config.namespace,
+    }, invoke_opts).value['result']
+    resources = _parse_yaml_document(objects, opts, config.transformations)
 
-    # Parse the manifest and create the specified resources.
-    resources = objects.apply(
-        lambda objects: _parse_yaml_document(objects, opts, config.transformations))
-
-    pulumi.Output.all(file, chart_dir, resources).apply(_cleanup_temp_dir)
     return resources
-
-
-def _fetch(chart: str, opts: FetchOpts) -> None:
```

Comment:
Are all of these fetch opts supported and exposed in the library?

IOW, do we have regression tests to validate the config options here still work for users?

I see we collect the JSON opts in `HelmFetchOpts` but `destination` and `verify` are missing

Reply:
All of the options are exposed that are still relevant. A few of the options were specific to v2 and not needed anymore.

The integration tests should hopefully catch regressions, although I don't think we cover every option.